src/test/java/com/exasol/releaserobot/github/GitHubPlatformValidatorTest.java

```diff
@@ -4,43 +4,37 @@
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertAll;
 import static org.mockito.Mockito.when;
 
 import java.util.*;
 
-import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
 
-import com.exasol.releaserobot.report.ValidationReport;
 import com.exasol.releaserobot.repository.*;
+import com.exasol.releaserobot.usecases.Report;
 
 class GitHubPlatformValidatorTest {
-    private ValidationReport validationReport;
-
-    @BeforeEach
-    void beforeEach() {
-        this.validationReport = new ValidationReport();
-    }
-
     @Test
     // [utest->dsn~validate-release-letter~1]
     void testValidateContainsHeader() {
         final ReleaseLetter changesLetter = Mockito.mock(ReleaseLetter.class);
         when(changesLetter.getHeader()).thenReturn(Optional.of("header"));
         final GitHubPlatformValidator validator = new GitHubPlatformValidator(null, null);
-        validator.validateContainsHeader(changesLetter, this.validationReport);
-        assertThat(this.validationReport.hasFailures(), equalTo(false));
+        final Report report = validator.validateContainsHeader(changesLetter);
+        assertThat(report.hasFailures(), equalTo(false));
     }
 
     @Test
     // [utest->dsn~validate-release-letter~1]
-    void testValidateDoesNotContainHeader() {
+    void testValidateContainHeaderFails() {
         final ReleaseLetter changesLetter = Mockito.mock(ReleaseLetter.class);
         when(changesLetter.getHeader()).thenReturn(Optional.empty());
         final GitHubPlatformValidator validator = new GitHubPlatformValidator(null, null);
-        validator.validateContainsHeader(changesLetter, this.validationReport);
-        assertThat(this.validationReport.getFailuresReport(), containsString("E-RR-VAL-1"));
+        final Report report = validator.validateContainsHeader(changesLetter);
```

Comment:
it would be good at some point in the future to add an adapter for decoupling the production code from the tests, so in future refactors we don't break the tests again :-) .

Same with removing references to private/protected methods and do everything through the public API of the production code.